book-1-orientation/chapters/ADONET_INTRO.md

```diff
@@ -9,624 +9,305 @@ This will be an interactive lesson. Your instructor will lead you through follow
+
+           //T his is to hold the actual foreign key integer
+           public int DepartmentId { get; set; }
+
+           // This property is for storing the C# object representing the department
+           public Department Department { get; set; }
+       }
+   }
+   ```
+
+1. Create a new folder called `Data`. This folder will contain classes that will be responsible for getting data out of our database and creating C# objects from that data. We typically call classes with this responsibility a Repository. Add two new files to it called `DepartmentRepository.cs` and `EmployeeRepository.cs`.
+1. Copy the following starter code into `DepartmentRepository.cs`
+
+   ```csharp
+   using System.Collections.Generic;
+   using System.Data.SqlClient;
+   using DepartmentsEmployees.Models;
+
+   namespace DepartmentsEmployees.Data
+   {
+       /// <summary>
+       ///  An object to contain all database interactions.
+       /// </summary>
+       public class DepartmentRepository
+       {
+           /// <summary>
+           ///  Represents a connection to the database.
+           ///   This is a "tunnel" to connect the application to the database.
+           ///   All communication between the application and database passes through this connection.
+           /// </summary>
+           public SqlConnection Connection
+           {
+               get
+               {
+                   // This is "address" of the database
+                   string _connectionString = "Data Source=localhost\\SQLEXPRESS;Initial Catalog=DepartmentsEmployees;Integrated Security=True";
+                   return new SqlConnection(_connectionString);
+               }
+           }
+
+       }
+   }
+
+   ```
+
+   The DepartmentRepository you just added contains a single, computed property called `Connection`. The type of this property is `SqlConnection`. It represents a connection from your C# application to your SQL Server database. Think of it like a two-way tunnel that all communication passes through. Since the property is computed, it means that any time the `Connection` property gets referenced in our code, it will create a new tunnel. Typically, this tunnel stays open only long enough to execute a single command (i.e. a `SELECT` or `INSERT` statement). Once the command is executed, we close the connection and effectively destroy that tunnel. Then when we want to execute another command, we do the same thing again and create a new tunnel.
+
+1. Let's see how we can get data out of our Department table and convert it into a `List<Department>`. Add the following method to your `DepartmentRepository` class
+
+   ```csharp
+   /// <summary>
+   ///  Returns a list of all departments in the database
+   /// </summary>
+   public List<Department> GetAllDepartments()
+   {
+       //  We must "use" the database connection.
+       //  Because a database is a shared resource (other applications may     be using it too) we must
```

Comment:
There's some odd spacing in the code snippets that begin here and continues to line 248.
```
...(other applications may     be using it too)...
```